packages/bpk-component-ticket/src/BpkTicket.js

```diff
@@ -16,132 +16,144 @@
  * limitations under the License.
  */
 
+/* @flow */
+
 import PropTypes from 'prop-types';
-import React from 'react';
+import React, { type Node } from 'react';
 import { cssModules } from 'bpk-react-utils';
 
 import STYLES from './bpk-ticket.scss';
 
 const getClassName = cssModules(STYLES);
 
-const isIE9 = () => {
-  if (typeof window === 'undefined') return false;
-  return window.navigator.appVersion.indexOf('MSIE 9.') !== -1;
+type Props = {
+  children: Node,
+  stub: Node,
+  stubProps: { [string]: any },
+  padded: boolean,
+  vertical: boolean,
+  withNotches: boolean,
+  className: ?string,
+  stubClassName: ?string,
+  href: ?string,
 };
 
-const BpkTicket = props => {
+const BpkTicket = (props: Props) => {
   const {
     children,
     href,
     padded,
     stub,
     vertical,
+    withNotches,
     className,
     stubClassName,
     stubProps,
     ...rest
   } = props;
 
-  const classNames = [getClassName('bpk-ticket')];
-  const mainClassNames = ['bpk-ticket__paper', 'bpk-ticket__main'].map(
-    getClassName,
+  const classNames = getClassName(
+    'bpk-ticket',
+    className,
+    vertical && 'bpk-ticket--vertical',
+    withNotches && 'bpk-ticket--with-notches',
   );
-  const mainInnerClassNames = [getClassName('bpk-ticket__main-inner')];
-  const stubClassNames = ['bpk-ticket__paper', 'bpk-ticket__stub'].map(
-    getClassName,
+
+  const mainClassNames = getClassName(
+    'bpk-ticket__paper',
+    'bpk-ticket__main',
+    padded && 'bpk-ticket__main--padded',
+    vertical && 'bpk-ticket__main--vertical',
+    !vertical && 'bpk-ticket__main--horizontal',
```

Comment:
What about `vertical ? 'bpk-ticket__main--vertical' : 'bpk-ticket__main--horizontal',`? It can be weird as it's the only place this would be used, but it also feels weird to have `vertical && ...` and in the next line `!vertical && ...` 😅